dri3_ws.c

```diff
@@ -820,15 +815,21 @@ static WSEGLError WSEGL_GetDrawableParameters(WSEGLDrawableHandle hDrawable,
 
 	DBG("drawable=%p, current-back=%u", drawable, drawable->current_back_idx);
 
-	if (!create_buffers(drawable)) {
-		if (drawable->drawable_type == DRI3WS_DRAWABLE_WINDOW )
-			return WSEGL_BAD_NATIVE_WINDOW;
-		else
-			return WSEGL_BAD_NATIVE_PIXMAP;
-	}
-
 	struct driws_buffer *buffer = drawable->buffers[drawable->current_back_idx];
 
+	if (!buffer) {
+		if (!create_buffers(drawable)) {
+			if (drawable->drawable_type == DRI3WS_DRAWABLE_WINDOW )
+				return WSEGL_BAD_NATIVE_WINDOW;
+			else
+				return WSEGL_BAD_NATIVE_PIXMAP;
+		}
+		buffer = drawable->buffers[drawable->current_back_idx];
+	} else {
+		if (drawable->size_changed)
+			return WSEGL_BAD_DRAWABLE;
+	}
```

Comment:
This might be slightly easier to read if you had
```
if (buffer && drawable->size_changed)
  return WSEGL_BAD_DRAWABLE;
```
before the create_buffers code.

Reply:
Updated